messages.rb

```diff
@@ -276,6 +289,92 @@ def send_list(event)
   event.attach_file(File::open(tmpfile))
 end
 
+def send_community(event)
+  msg = event.content
+  tabs = parse_tabs(msg)
+  tabs.empty? ? ftabs = "" : ftabs = format_tabs(tabs)
+  tabs = (tabs.empty? ? [:SI, :S, :SL, :SU, :SS, :SS2] : tabs)
+
+  levels = Score.where(rank: 0, highscoreable_type: "Level").includes(:level).where(levels: {tab: tabs})
+  episodes = Score.where(rank: 0, highscoreable_type: "Episode").includes(:episode).where(episodes: {tab: tabs})
+  lvls = levels.count
+  eps = episodes.count
+  tls = levels.map{|s| s.score}.sum
+  tes = episodes.map{|s| s.score}.sum
+  if !(tabs&[:SS, :SS2]).empty? # in order to compare lvls and eps we have to exclude secrets
+    tlsno = Score.where(rank: 0, highscoreable_type: "Level").includes(:level).where(levels: {tab: tabs}).where.not(levels: {tab: [:SS, :SS2]}).map{|s| s.score}.sum
+    ttlsno = "Total level score (w/o secrets): #{tlsno}\n"
+  else
+    tlsno = tls
+    ttlsno = ""
+  end
+  diff = tlsno-4*90*eps-tes
```

Comment:
I'm a bit confused about what this difference is computing exactly. What's the goal here?

Reply:
Oh yeah, the goal is to see how much time is being lost in the episode 0ths with respect to the level 0ths, both in total, and later on also in average (so you can see how much time on average is lost per episode). In other words, how clean episodes are globally. This function was also present in NHigh (albeit less complete).

However, we have to take into account that each level gives you 90 seconds at the start, but when playing an episode, only the first level actually does this, and you have to play the remaining levels with the time you've accumulated up to that point, as of course, you already know.

This means that the episode score will always be 4 * 90 seconds lower than the sum of all 5 level scores, so if you want to compare both (e.g. to see how clean the episode is), you will have to subtract that amount.

Naturally this has the same effect when we find the total sum, so we will have to subtract that amount times the number of episodes there are. This is how we find the true difference, which we can later average as well :)

Reply:
Ah, that makes sense, cool. Maybe a quick comment here explaining, and you may want to add a bit of explanation to the output as well (instead of just 'Average difference:' which is a bit opaque) - though I'm having a hard time thinking of how you would describe this without being way too wordy...

'Average difference between level and episode 0ths:' maybe? Even that's pretty long, but it at least attempts to explain what's being computed.

Reply:
True, clarified the code and output.